src/components/modal/modal.tsx

```diff
@@ -0,0 +1,65 @@
+import { Component, ComponentInterface, h, Method, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'bds-modal',
+  styleUrl: 'modal.scss',
+  shadow: true,
+})
+export class BdsModal implements ComponentInterface {
+  /**
+   * Used to open/close the modal
+   */
+  @Prop({
+    mutable: true,
+    reflect: true,
+  })
+  public open?: boolean = false;
+
+    /**
+   * Used to open/close the modal
+   */
+  @Prop({
+    mutable: true,
+    reflect: true,
+  })
+  public closeButton?: boolean = true;
+
+  /**
+   * Used to show or hide close button
+   */
+  @Prop({
+    mutable: true,
+    reflect: true,
+  })
+
+  /**
+   * Can be used outside to open/close the modal
+   */
+  @Method()
+  async toggle() {
+    this.open = !this.open;
+  }
+
+  private handleMouseClick = (): void => {
+    this.open = false;
```

Comment:
I think maybe it would be a good idea if you add an event emitter to tell when the user clicked on the icon to close the modal